sdks/java/container/build.gradle

```diff
@@ -68,6 +73,28 @@ golang {
   }
 }
 
+// this is a workaround to call generateLicenseReport task from project root directory.
+// generateLicenseReport does not return correct dependency list when not called from the root.
+task generateThirdPartyLicenses(type: Exec) {
+  workingDir project.rootProject.projectDir
+  commandLine './sdks/java/container/license_scripts/license_script.sh'
+}
+
+//command: ./gradlew generateLicenseReport
+import com.github.jk1.license.render.*
+licenseReport {
+  outputDir = 'java_third_party_licenses'
+  projects = [project.rootProject] + project.rootProject.subprojects
+  excludeOwnGroup = true
+  configurations = ALL
```

Comment:
Why ALL? Should this just be `runtime`?

This pulls in lots of configurations that only are used to run static analysis at build time, etc, and do not affect the license of the container.

Reply:
I recently hit a problem where dependency resolution here is incorrect for one of these configurations. It is pulling versions that don't exist and nothing depends on.

Reply:
I tried with `runtime` at first, and it detected way less dependencies than `ALL`. To be safe (with legal issues), I changed the setting to `ALL`.

I noticed that multiple different versions are detected for some dependencies. I checked several of them and they all looked correct. Does the issue go away if you change the setting to `runtime`? How did you notice the issue? (the licenses were not pulled automatically so you have to add links manually?) I am asking this to see if there is an easy way to check unrelated dependencies/versions are detected.

Reply:
The issue I hit was https://github.com/kelloggm/checkerframework-gradle-plugin/issues/109 where the plugin creates a broken configuration that is actually not needed.